gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -341,6 +360,19 @@ private void onApplyWindowInsets(WindowInsetsCompat insets) {
     updateBottomSheetPeekHeight(insets);
   }
 
+  private void onAddPolygon(Point point) {
+    if (polygonPoints.size() > 3) {
+      if (polygonPoints.contains(point)) {
+        //viewModel.addPolygon(project, layer, polygonPoints);
+        mapContainerViewModel.setViewMode(Mode.DEFAULT);
+      }
+      polygonPoints.add(point);
```

Comment:
Since this second part is repeated in both branches, it can be moved outside the `if` condition, and the `else` removed.